packages/readable-js-assets-webpack-plugin/index.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * External dependencies
+ */
+const fs = require( 'fs' );
+const path = require( 'path' );
+
+class AddReadableJsAssetsWebpackPlugin {
+	extractUnminifiedFiles( compilation ) {
+		const files = compilation.chunks.flatMap( ( chunk ) => chunk.files );
+		compilation.unminifiedAssets = files.map( ( file ) => {
+			const asset = compilation.assets[ file ];
+			const unminifiedFile = file.replace( /\.min\.js$/, '.js' );
+			return [ unminifiedFile, asset.source() ];
+		} );
+	}
+	async writeUnminifiedFiles( compilation ) {
+		for ( const [ file, source ] of compilation.unminifiedAssets ) {
+			await fs.promises.writeFile(
+				path.join( compilation.options.output.path, file ),
```

Comment:
`compiler.outputFileSystem.writeFile` and `.join` could be used here. Unfortunately `.writeFile` doesn't support promises there.

The Jest test could then use an in-memory FS implementation. But rewriting the test is more like a fun project to play with, not necessary to get this merged.